verification/policy.go

```diff
@@ -0,0 +1,132 @@
+/*
+Package verification provides the utilities for handling verification related logic
+like Trust Stores and Trust Policies. Few utilities include loading, parsing, and validating
+trust policies and trust stores.
+*/
+package verification
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+// PolicyDocument represents a trustPolicy.json document
+type PolicyDocument struct {
+	// Version of the policy document
+	Version string `json:"version"`
+	// TrustPolicies include each policy statement
+	TrustPolicies []TrustPolicy `json:"trustPolicies"`
+}
+
+// TrustPolicy represents a policy statement in the policy document
+type TrustPolicy struct {
+	// Name of the policy statement
+	Name string `json:"name"`
+	// RegistryScopes that this policy statement affects
+	RegistryScopes []string `json:"registryScopes"`
+	// SignatureVerification setting for this policy statement
+	SignatureVerification string `json:"signatureVerification"`
+	// TrustStore this policy statement uses
+	TrustStore string `json:"trustStore,omitempty"`
+	// TrustedIdentities this policy statement pins
+	TrustedIdentities []string `json:"trustedIdentities,omitempty"`
+}
+
+func isPresent(val string, values []string) bool {
+	for _, v := range values {
+		if v == val {
+			return true
+		}
+	}
+	return false
+}
+
+// ValidatePolicyDocument validates a policy document according to it's version's rule set.
+// if any rule is violated, returns an error
+func ValidatePolicyDocument(policyDoc *PolicyDocument) error {
+	// Constants
+	wildcard := "*"
+	supportedPolicyVersions := []string{"1.0"}
+	supportedVerificationPresets := []string{"strict", "permissive", "audit", "skip"}
+	supportedTrustStorePrefixes := []string{"ca"}
```

Comment:
Would be nice to change these into types in next PR or when we implement verification logic. The verification logic will refer these values.

See https://github.com/notaryproject/notation-go/blob/main/plugin/errors.go#L9 as an example.